eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -105,12 +105,28 @@ object InteractiveTxBuilder {
                                  isInitiator: Boolean,
                                  localAmount: Satoshi,
                                  remoteAmount: Satoshi,
+                                 commonInput_opt: Option[InputInfo],
                                  fundingPubkeyScript: ByteVector,
                                  lockTime: Long,
                                  dustLimit: Satoshi,
                                  targetFeerate: FeeratePerKw,
-                                 requireConfirmedInputs: RequireConfirmedInputs) {
-    val fundingAmount: Satoshi = localAmount + remoteAmount
+                                 requireConfirmedInputs: RequireConfirmedInputs,
+                                 localSpliceOut_opt: Option[TxOut] = None,
+                                 remoteSpliceOut_opt: Option[TxOut] = None) {
+    val previousAmount: Satoshi = commonInput_opt.map(_.txOut.amount).getOrElse(0 sat)
+
+    val localSpliceOutAmount: Satoshi = localSpliceOut_opt.map(_.amount).getOrElse(0 sat)
+    val remoteSpliceOutAmount: Satoshi = remoteSpliceOut_opt.map(_.amount).getOrElse(0 sat)
+    // splice-out are special, because there is potentially no additional funding, so the mining fees must be taken from the previous funding amount
+    // if there is one splice-out, the side that does it pays; if both sides do a splice-out, the initiator pays
+    // TODO: weight is hardcoded and is the same whether there are one or two splices, it should be fully computed using the actual weight of each splice output
+    val localSpliceOutFee: Satoshi = if (localSpliceOut_opt.isDefined && (remoteSpliceOut_opt.isEmpty || isInitiator)) Transactions.weight2fee(targetFeerate, 600) else 0 sat
+    val remoteSpliceOutFee: Satoshi = if (remoteSpliceOut_opt.isDefined && (localSpliceOut_opt.isEmpty || !isInitiator)) Transactions.weight2fee(targetFeerate, 600) else 0 sat
```

Comment:
I think it should be fairly easy to compute the weight of the splice tx (assuming there is no splice-in), right? Then we could do a fairer split if both sides do a splice-out.